builtin/credential/aws/path_role.go

```diff
@@ -320,6 +322,142 @@ func (b *backend) setRole(ctx context.Context, s logical.Storage, roleName strin
 	return nil
 }
 
+// initialize is used to initialize the AWS roles
+func (b *backend) initialize(ctx context.Context, s logical.Storage) error {
+
+	// Initialize only if we are either:
+	//   (1) A local mount.
+	//   (2) Are _not_ a replicated standby cluster.
+	if b.System().LocalMount() || !b.System().ReplicationState().HasState(consts.ReplicationPerformanceSecondary|consts.ReplicationPerformanceStandby) {
+
+		logger := b.Logger().Named("initialize")
+
+		// grab the guard
+		if !atomic.CompareAndSwapUint32(b.initializeCASGuard, 0, 1) {
+			logger.Error("initialization already in progress")
+			return nil
+		}
+
+		// check if this method has already been called
+		if b.isInitialized {
+			atomic.StoreUint32(b.initializeCASGuard, 0)
+			logger.Error("already initialized")
+			return nil
+		}
+		b.isInitialized = true
+
+		// kick off the role upgrader
+		go func() {
+			defer atomic.StoreUint32(b.initializeCASGuard, 0)
+
+			// Don't cancel when the original client request goes away
+			ctx = context.Background()
+
+			logger.Info("starting initialization")
+
+			updated, err := b.updateUpgradableRoleEntries(ctx, s)
+			if err != nil {
+				logger.Error("error running initialization", "error", err)
+				return
+			}
+			if !updated {
+				logger.Info("upgrade has already been performed for role storage version",
+					currentRoleStorageVersion)
+			}
+
+			logger.Info("initialization succeeded")
+		}()
+
+		return nil
+	}
+
+	return nil
+
+}
+
+// updateUpgradableRoleEntries upgrades and persists all of the role entries
+// that are in need of being upgraded.
+func (b *backend) updateUpgradableRoleEntries(ctx context.Context, s logical.Storage) (bool, error) {
+
+	// check if we've already upgraded to the current role storage
+	// version
+	version, err := b.upgradedRoleStorageVersion(ctx, s)
+	if err != nil {
+		return false, err
+	}
+	if version == currentRoleStorageVersion {
```

Comment:
Please don't store a raw integer. Use a struct with the version as a member. If it makes sense, piggy back on one of the other config structs that exist and simply add the backend version to that.